src/helpers/createPost.ts

```diff
@@ -15,21 +16,33 @@ export default async function (
   replyToId?: string
 ) {
   TextFormStore.loading = true
+  TextFormStore.error = null
 
   let currentAccount
   const { privateKey } = BurnerWalletStore
-  const { account } = walletStore
+  let { account } = walletStore
 
   if (privateKey) currentAccount = new Wallet(privateKey).address
   if (account && (await hasFarcasterBadge(account))) currentAccount = account
 
-  TextFormStore.error = null
-
   try {
     if (!currentAccount) {
-      TextFormStore.waitBurner = true
-      return
+      if (!account) account = await walletStore.connect(true)
+      if (!account) throw 'Please, connect an account with badge'
+
+      const newPrivateKey = await BurnerWalletStore.generateBurnerWallet(
+        account
+      )
+      if (!newPrivateKey)
+        throw 'An error occurred while creating the burner wallet, please try again'
```

Comment:
The error looks similar to the one on line 44. Can we combine all the `throw errors` in one?